### ISSUE TITLE
Add an on-screen numpad for use on any smartphone

### ISSUE THREAD

**Issue by sspanak** [sspanak/tt9#159]
There is no good way of using Traditional T9 on a touchscreen-only device. So let's add an on-screen numpad to make it usable.

## Settings screen
- Change the "On-screen keys" option to toggle the soft numpad.
- Rename it to "On-screen keypad"
- Make it a dropdown with three options: "Disabled", "Control keys only", "Full-sized".

## Design 
The layout is similar to Gboard's GODAN.

```
===================================================
| suggestion | are | taller | like | in | Gboard |
===================================================
|    <    |  Clear  | Filter_ |    >    |
==================================================================================
|               |   1           |   2           |   3           |       ⌫       |
|              	|       ,.?!()  |       abc     |       def     |               |
+---------------+---------------+---------------+---------------+---------------+
|               |   4           |   5           |   6           |      OK       |
|               |       ghi     |       jkl     |       mno     |               |
+---------------+---------------+---------------+---------------+---------------+
|       ⚙      |   7           |   8           |   9           |      🪐       |
|               |       pqrs    |       tuv     |       wxyz    |               |
+---------------+---------------+---------------+---------------+---------------+
|  +Add word    |   *           |   0           |   #           | ⌨️ T9/ABC/123 |
|               |               |   ␣ / +%*{}   |               | ⇧ Shift       |
+---------------+---------------+---------------+---------------+---------------+


```

- Suggestions row
    - Contains the suggestion list
    - Is as tall as in Gboard
- Suggestion controls row
    - Button meanings: Previous suggestion | Clear stem filter | Set stem filter | Next suggestion
    - "Filter" is in bold+italic, but the underscore is not.
- Left control column
    - Button meanings (bottom to top): Add Word, Show Settings
    - Consider putting only an icon for "Add Word".
    - Consider adding a "left arrow" for moving the cursor to the left. (after #157)
- Right control column
    - Button meanings (bottom to top): Input Mode, Next Language, OK, backspace
    - When the mode changes, the letters on the numpad have their text case (or language) changed accordingly.
    - Consider adding a "right arrow" for moving the cursor to the right. (after #157)
- Numpad
    - If there is enough room put the number on the left and the characters on the right. If not, put a small number in the upper left corner.
    - The character list could be centered or aligned to the right with some padding
    - 2-9:
        - First five characters for this key are displayed
        - Button text case changes according to the current state
    - 0 and 1:
        - Either display the first five or a meaningful hardcoded list
    - `#` and `*`: Just type the respective character
    - "Upside-Down Keypad" flips the number keys upside down.

## Refactoring
- Create a `handlers` directory inside `ime`.
    - `ime/KeyPadHandler` -> `ime/input/KeyPadHandler`
    - `ime/SoftKeyHandler` -> `ime/input/SoftControlPadHandler`
    - create `ime/input/SoftNumPadHandler

## Second phase
These should probably become separate issues.

- Vibrate on keypress
- Suggestions row
  - Touching a suggestion selects it
  - ~~Sliding, scrolls them~~ - the RecyclerView supports it by default
- Right control column
    - _(Not sure about this yet)_ Input Mode shows the current mode: T9, ABC, abc or 123 (or maybe the next mode?).
    - _(Not sure about this yet)_ Consider changing Input Mode icon and text to `⇧ Shift`, when composing a word.

**Comment by sspanak** [comment]
Took a slightly different approach with the layout and everything. 

There remaining stuff is:
- Restore the key inscriptions.
- 123 mode does not work at all.
- Verify the candidate colors look fine in the small main menu.
- Verify the small menu is still working fine in general.

Do these in a follow-up issue:
- Vibration on key press (+ a setting)
- Add Filter/Clear filter keys
- Move left/Move right after or in #157
- More meaningful icons or text for the control keys (Add word, mode, etc).
- `*` and `#` change to `?` and `!` in non-numeric modes.
- Add proper support for the "upside-down keyboard" setting.
